app/DTO/AlbumProtectionPolicy.php

```diff
@@ -2,50 +2,71 @@
 
 namespace App\DTO;
 
-class AlbumProtectionPolicy extends DTO
-{
-	public const IS_PUBLIC_ATTRIBUTE = 'is_public';
-	public const REQUIRES_LINK_ATTRIBUTE = 'requires_link';
-	public const IS_NSFW_ATTRIBUTE = 'is_nsfw';
-	public const IS_DOWNLOADABLE_ATTRIBUTE = 'is_downloadable';
-	public const IS_SHARE_BUTTON_VISIBLE_ATTRIBUTE = 'is_share_button_visible';
-	public const GRANTS_FULL_PHOTO_ATTRIBUTE = 'grants_full_photo';
-
-	public bool $isPublic;
-	public bool $requiresLink;
-	public bool $isNSFW;
-	public bool $isDownloadable;
-	public bool $isShareButtonVisible;
-	public bool $grantsFullPhoto;
+use App\Models\BaseAlbumImpl;
+use App\Models\Extensions\BaseAlbum;
+use App\SmartAlbums\BaseSmartAlbum;
 
+/**
+ * This represents the Album Protection policy.
+ *
+ * In other words it provides information on the security attributes of an album.
+ * It MUST not be used in the front-end to determine whether an action is permitted or not (e.g. share link available).
+ *
+ * It is used in two places:
+ * 1. When sent to the front-end:
+ *   - provides an overview of the policy of the current visited album
+ *   - allows modification of the policy on non-smart albums
+ * 2. When gotten from the front-end:
+ *   - to allow easy interface between the validated request {@link \App\Http\Requests\Album\SetAlbumProtectionPolicyRequest}
+ *     and the applied action {@link \App\Actions\Album\SetProtectionPolicy}.
+ */
+class AlbumProtectionPolicy extends ArrayableDTO
+{
 	public function __construct(
-		bool $isPublic,
-		bool $requiresLink,
-		bool $isNSFW,
-		bool $isDownloadable,
-		bool $isShareButtonVisible,
-		bool $grantsFullPhoto
+		public bool $is_public,
+		public bool $is_link_required,
+		public bool $is_nsfw,
+		public bool $grants_access_full_photo,
+		public bool $grants_download,
+		public bool $is_password_required = false, // Only used when sending info to the front-end
 	) {
-		$this->isPublic = $isPublic;
-		$this->requiresLink = $requiresLink;
-		$this->isNSFW = $isNSFW;
-		$this->isDownloadable = $isDownloadable;
-		$this->isShareButtonVisible = $isShareButtonVisible;
-		$this->grantsFullPhoto = $grantsFullPhoto;
 	}
 
 	/**
-	 * {@inheritDoc}
+	 * Given a BaseAlbum, returns the Protection Policy associated to it.
+	 *
+	 * @param BaseAlbum|BaseAlbumImpl $baseAlbum
+	 *
+	 * @return AlbumProtectionPolicy
+	 */
+	public static function ofBaseAlbum(BaseAlbum|BaseAlbumImpl $baseAlbum): AlbumProtectionPolicy
+	{
+		return new AlbumProtectionPolicy(
+			is_public: $baseAlbum->is_public,
+			is_link_required: $baseAlbum->is_link_required,
+			is_nsfw: $baseAlbum->is_nsfw,
+			grants_access_full_photo: $baseAlbum->grants_access_full_photo,
```

Comment:
```suggestion
			grants_full_photo_access: $baseAlbum->grants_full_photo_access,
```

I won't comment this one any more times as this had already been resolved in the old PR and it is much easier to refactor this with a proper IDE than changing every occurrence via Github comments.

Reply:
Fixed [31a5e2b](https://github.com/LycheeOrg/Lychee/pull/1539/commits/31a5e2b97c5302595e7b13326c54d6980c16498b)